source/detail/environment.hpp

```diff
@@ -63,11 +65,26 @@
 
 /// <summary>
 /// Returns whether the C version `version` is supported by the current build configuration.
+/// Unfortunately, while __STDC_VERSION__ must be defined in C compilers, when using C++ compilers it is
+/// implementation-defined whether the __STDC_VERSION__ is defined or not. However, the C++ standard defines
+/// for each C++ version which C version it refers to. This can be found out by using the following list:
+/// C++98 -> C95
+/// C++11 -> C99
+/// C++17 -> C11
+/// C++20 -> C17
+/// C++26 -> C23
 /// </summary>
 /// <seealso cref="XLNT_C_11">
 /// <seealso cref="XLNT_C_17">
 /// <seealso cref="XLNT_C_23">
-#define XLNT_HAS_C_VERSION(version) (1/version == 1/version && XLNT_C_VERSION >= version)
+#define XLNT_HAS_C_VERSION(version) (1/version == 1/version && \
+    ((defined(__STDC_VERSION__) && __STDC_VERSION__ >= version) || \
+     (version == XLNT_C_11 && XLNT_HAS_CPP_VERSION(XLNT_CPP_17)) || \
```

Comment:
Is it a good idea to use an equality operator to compare versions? May one use an intermediate number (in rare cases)?

Reply:
No, it's a very bad one. Good catch thank you! :+1: :smile: I fixed it in my latest commits. The previous code also wouldn't have detected C11 if C17 was also available, which was simply wrong.